drivers/spi/spi_max32.c

```diff
@@ -565,6 +629,99 @@ static int transceive_dma(const struct device *dev, const struct spi_config *con
 }
 #endif /* CONFIG_SPI_MAX32_DMA */
 
+#ifdef CONFIG_SPI_RTIO
+static void spi_max32_iodev_complete(const struct device *dev, int status);
+
+static void spi_max32_iodev_start(const struct device *dev)
+{
+	struct max32_spi_data *data = dev->data;
+	struct spi_rtio *rtio_ctx = data->rtio_ctx;
+	struct rtio_sqe *sqe = &rtio_ctx->txn_curr->sqe;
+	int ret = 0;
+
+	switch (sqe->op) {
+	case RTIO_OP_RX:
+	case RTIO_OP_TX:
+	case RTIO_OP_TINY_TX:
+	case RTIO_OP_TXRX:
+		ret = spi_max32_transceive(dev);
```

Comment:
transceive might be a bit cleaner if given the buffer pair to work on, given NULL for tx or rx if unidirection perhaps.

Reply:
Since initially transceive is able to handle all the cases, i didn't want to change core functionality of the drive, but i think now this is covered with previous comment code update